src/Orleans/Streams/SimpleMessageStream/SimpleMessageStreamProducerExtension.cs

```diff
@@ -42,7 +44,7 @@ internal void AddStream(StreamId streamId)
             // so this call is only made once, when StreamProducer is created.
             if (remoteConsumers.TryGetValue(streamId, out obs)) return;
 
-            obs = new StreamConsumerExtensionCollection(streamPubSub, logger);
+            obs = new StreamConsumerExtensionCollection(streamPubSub, providerRuntime.ServiceProvider.GetRequiredService<ILoggerFactory>());
             remoteConsumers.Add(streamId, obs);
```

Comment:
loggerFactory is being injected in the constructor already, so just pass that one

Reply:
I will be more careful next time.